src/block/mod.rs

```diff
@@ -214,13 +209,47 @@ pub fn get_blockchain_length(file_name: &OsStr) -> Option<u32> {
 }
 
 pub fn is_valid_block_file(path: &Path) -> bool {
-    if let Some(file_name) = path.file_name() {
-        get_state_hash(file_name).is_some()
-            && file_name
-                .to_str()
-                .map(|file_name| file_name.ends_with(".json"))
-                .unwrap_or(false)
+    fn is_valid_state_hash(input: &str) -> bool {
+        input.starts_with("3N") && input.len() == 52
+    }
+    if let Some(ext) = path.extension() {
+        // check json extension
+        if ext.to_str() == Some("json") {
+            // check file stem
+            if let Some(file_name) = path.file_stem() {
+                if let Some(parts) = file_name
+                    .to_str()
+                    .map(|name| name.split('-').collect::<Vec<&str>>())
+                {
+                    if parts.len() == 2 {
```

Comment:
is there anyway use `.last()` for `Vec`? Might simplify this. I'm not blocking the MR on this.

Reply:
Great suggestion, much simpler that way